addons/l10n_jo_edi/models/account_edi_xml_ubl_21_jo.py

```diff
@@ -0,0 +1,302 @@
+from lxml import etree
+
+from odoo import models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_repr
+from odoo.tools.float_utils import float_round
+
+
+JO_CURRENCY = type('JoCurrency', (), {'name': 'JO'})()
+
+
+class AccountEdiXmlUBL21JO(models.AbstractModel):
+    _name = "account.edi.xml.ubl_21.jo"
+    _inherit = 'account.edi.xml.ubl_21'
+    _description = "UBL 2.1 (JoFotara)"
+
+    ####################################################
+    # overriding vals methods of account.edi.xml.ubl_20
+    ####################################################
+
+    def _get_country_vals(self, country):
+        return {
+            'identification_code': country.code,
+        }
+
+    def _get_partner_party_identification_vals_list(self, partner):
+        if partner.vat and not partner.vat.isdigit():
+            raise ValidationError("JoFotara portal cannot process customer VAT with non-digit characters in it")
+        return [{
+            'id_attrs': {'schemeID': 'TN'},
+            'id': partner.vat,
+        }]
+
+    def _get_partner_address_vals(self, partner):
+        return {
```

Comment:
@smetl I am overriding almost all methods, because the Jordanian documentation describes pretty different XMLs from the ubl standard. That is why I divided the file into sections (e.g., one of the sections contains all methods overriden from `account.edi.xml.ubl_20` model). Moreover, the explanation for overriding all the methods is pretty much the same: following the XML file specifications from the Jordanian tax authority documentation.